manifest/openapi/schema.go

```diff
@@ -66,6 +66,20 @@ func getTypeFromSchema(elem *openapi3.Schema, stackdepth uint64, typeCache *sync
 
 	var t tftypes.Type
 
+	// Check if attribute type is tagged as 'x-kubernetes-preserve-unknown-fields' in OpenAPI.
+	// If so, we add a type hint to indicate this and return DynamicPseudoType for this attribute,
+	// since we have no further structural information about it.
+	if xpufJSON, ok := elem.Extensions["x-kubernetes-preserve-unknown-fields"]; ok {
```

Comment:
We use the string literal `"x-kubernetes-preserve-unknown-fields"` in a bunch of places now, I wonder if we should have a const for it?